src/main/java/ghebrat/bes/theresFoodAtHome/user/UserController.java

```diff
@@ -25,4 +25,19 @@ public UserController(UserService userService) {
     public void createUser(@RequestBody User user) throws UserException {
         userService.addUser(user);
     }
+
+    @PutMapping("/{userId}")
+    public void UpdateUser(@PathVariable Integer userId,
+                           @RequestParam(required = false) String firstName,
```

Comment:
I would set all of theres to required and then you can remove `firstName != null` in UserService updateUser